cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AuthorizationModule.java

```diff
@@ -122,7 +122,7 @@ public DynamicDatasetCache get() {
       return new MultiThreadDatasetCache(
         dsInstantiator, txClient, NamespaceId.SYSTEM, ImmutableMap.<String, String>of(),
         metricsCollectionService.getContext(ImmutableMap.<String, String>of()),
-        ImmutableMap.<String, Map<String, String>>of()
+        ImmutableMap.<String, Map<String, String>>of(), "Authorization"
```

Comment:
The client id should represent the process that is making the transaction calls. I think it would be good to create a wrapper for `TransactionSystemClient` that has the client id. This wrapper client will have to be created only once during the startup, and other code will not have to change to accommodate client id.